ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -298,45 +298,51 @@ onnxruntime_subplugin::configure_instance (const GstTensorFilterProperties *prop
-    inputNode.names.push_back (inputNode.names_allocated_strings.back ().get ());
-
-    /* Get input type and shape */
-    Ort::TypeInfo type_info = session.GetInputTypeInfo (i);
-    auto tensor_info = type_info.GetTensorTypeAndShapeInfo ();
-    inputNode.types.push_back (tensor_info.GetElementType ());
-    inputNode.shapes.push_back (tensor_info.GetShape ());
-  }
+  try {
+    Ort::AllocatorWithDefaultOptions allocator;
+
+    /* Initialize input info */
+    inputNode.count = num_inputs;
+
+    for (i = 0; i < num_inputs; i++) {
+      /* Get input name */
+      auto input_name = session.GetInputNameAllocated (i, allocator);
+      inputNode.names_allocated_strings.push_back (std::move (input_name));
+      inputNode.names.push_back (inputNode.names_allocated_strings.back ().get ());
+
+      /* Get input type and shape */
+      Ort::TypeInfo type_info = session.GetInputTypeInfo (i);
+      auto tensor_info = type_info.GetTensorTypeAndShapeInfo ();
+      inputNode.types.push_back (tensor_info.GetElementType ());
+      inputNode.shapes.push_back (tensor_info.GetShape ());
+    }
 
-  /* Initialize output info */
-  outputNode.count = num_outputs;
+    /* Initialize output info */
+    outputNode.count = num_outputs;
 
-  for (i = 0; i < num_outputs; i++) {
-    /* Get output name */
-    auto output_name = session.GetOutputNameAllocated (i, allocator);
-    outputNode.names_allocated_strings.push_back (std::move (output_name));
-    outputNode.names.push_back (outputNode.names_allocated_strings.back ().get ());
+    for (i = 0; i < num_outputs; i++) {
+      /* Get output name */
+      auto output_name = session.GetOutputNameAllocated (i, allocator);
+      outputNode.names_allocated_strings.push_back (std::move (output_name));
+      outputNode.names.push_back (outputNode.names_allocated_strings.back ().get ());
 
-    /* Get output type and shape */
-    Ort::TypeInfo type_info = session.GetOutputTypeInfo (i);
-    auto tensor_info = type_info.GetTensorTypeAndShapeInfo ();
-    outputNode.types.push_back (tensor_info.GetElementType ());
-    outputNode.shapes.push_back (tensor_info.GetShape ());
-  }
+      /* Get output type and shape */
+      Ort::TypeInfo type_info = session.GetOutputTypeInfo (i);
+      auto tensor_info = type_info.GetTensorTypeAndShapeInfo ();
+      outputNode.types.push_back (tensor_info.GetElementType ());
+      outputNode.shapes.push_back (tensor_info.GetShape ());
+    }
+
+    memInfo = Ort::MemoryInfo::CreateCpu (
+        OrtAllocatorType::OrtArenaAllocator, OrtMemType::OrtMemTypeDefault);
 
-  memInfo = Ort::MemoryInfo::CreateCpu (
-      OrtAllocatorType::OrtArenaAllocator, OrtMemType::OrtMemTypeDefault);
+    configured = true;
+    allocator = Ort::AllocatorWithDefaultOptions{ nullptr }; /* delete unique_ptr */
 
-  configured = true;
-  allocator = Ort::AllocatorWithDefaultOptions{ nullptr }; /* delete unique_ptr */
+  } catch (...) {
+    cleanup ();
```

Comment:
Cleanup() has no effect at error handling because configured may be false.
(```if configured == false```, cleanup() does NOTHING.)

Reply:
I added a `force` parameter with a default value of `false` to maintain backward compatibility.
I modified the catch block to call `cleanup(true)` for proper resource cleanup.